framework/test/unit/modules/interoperability/internal_method.spec.ts

```diff
@@ -1677,6 +1943,17 @@ describe('Base interoperability internal method', () => {
 				certificateSchema,
 				expect.toBeObject() as Certificate,
 			);
+
+			// TODO [DONE]: checked that calculateRootFromRightWitness is called with the correct arguments.
+			expect(regularMerkleTree.calculateRootFromRightWitness).toHaveBeenCalledWith(
+				updatedInboxTree.size,
+				updatedInboxTree.appendPath,
+				crossChainUpdateParams.inboxUpdate.messageWitnessHashes,
+			);
 		});
+
+		// TODO: There should be some test(s) where inboxUpdate.crossChainMessages is non empty
+		// TODO: checked that regularMerkleTree.calculateMerkleRoot is called for every ccm and that it is called with the correct arguments
+		// TODO: i.e. calculateMerkleRoot must be called two times
```

Comment:
this is still open

Reply:
Yes, I have raised a question above regarding to this one

> 🔴 From this [test](https://github.com/LiskHQ/lisk-sdk/blob/6c4d1c7d44b0f1199754794fa9be0f2ed210df91/framework/test/unit/modules/interoperability/internal_method.spec.ts#L1908C32-L1908C32), I can see calculateMerkleRoot being called once only. Not sure if the approach has been changed if I am not testing it properly, advice is appreciated.

Reply:
It took me also a bit to recall what I meant :) It is about [this step in the LIP](https://github.com/LiskHQ/lips/blob/8c374e7649c48bcc8bf9015ca2ebc8272b72998d/proposals/lip-0053.md?plain=1#L303) or [this one in the code](https://github.com/LiskHQ/lisk-sdk/blob/0aabfa637aa2b37b4a84b900dbaf92feacb107e2/framework/src/modules/interoperability/base_interoperability_internal_methods.ts#L681-L685).